src/main/java/io/github/genomicdatainfrastructure/discovery/filters/infrastructure/ckan/CkanFilterBuilder.java

```diff
@@ -172,13 +179,9 @@ private FilterRange extractDateTimeRange(CkanFacet facet) {
         var min = bounds.stream().min(Comparator.naturalOrder()).orElse(null);
         var max = bounds.stream().max(Comparator.naturalOrder()).orElse(null);
```

Comment:
```suggestion
        var min = bounds.stream().min(Comparator.naturalOrder()).orElseThrow();
        var max = bounds.stream().max(Comparator.naturalOrder()).orElseThrow();
```

Reply:
Since `bounds` is never empty, the `min` could never return an empty `Optional`. Same for `max`.